metadata-ingestion/src/datahub/ingestion/extractor/schema_util.py

```diff
@@ -71,7 +71,7 @@ def avro_schema_to_mce_fields(avro_schema_string: str) -> List[SchemaField]:
             type=_get_column_type(parsed_field.type),
             description=parsed_field.props.get("doc", None),
             recursive=False,
-            nullable=(parsed_field.type == "null"),
+            nullable=parsed_field.has_default,
```

Comment:
Fields can also be null if they are of type [ "null", "something else" ] as discussed [here](https://stackoverflow.com/questions/29299610/is-it-possible-to-have-an-optional-field-in-an-avro-schema-i-e-the-field-does). This change will miss those cases, right?

Reply:
For full correctness, the `nullable` logic might need to be a bit more complex: union types should inherit their default values and nullability from the first type in the union.

This might be ok to do in a follow-up PR

Reply:
Thanks, my avro knowledge is far from good. I updated the PR to a solution that more closely resembles the existing behaviour. Please let me know what you think.